VContainer/Assets/VContainer/Runtime/Unity/ObjectResolverUnityExtensions.cs

```diff
@@ -33,7 +33,20 @@ void InjectGameObjectRecursive(GameObject current)
 
         public static T Instantiate<T>(this IObjectResolver resolver, T prefab) where T : UnityEngine.Object
         {
-            var instance = UnityEngine.Object.Instantiate(prefab);
+            var scope = (LifetimeScope)resolver.ApplicationOrigin;
+
+            T instance;
+            if (scope.IsRoot)
+            {
+                instance = UnityEngine.Object.Instantiate(prefab);
+                UnityEngine.Object.DontDestroyOnLoad(instance);
```

Comment:
It is intuitive, but it is a slightly different matter and a minor breaking change.

Instead of doing this, why not make the API explicit in the Register declaration?

like this:
`RegisterComponentInNewPrefab(...).DontDestroyOnLoad()`

Reply:
I have reconsidered and will adopt this code as it is nonsense to Instantiate the root scope to the first scene.